indico/modules/categories/models/categories.py

```diff
@@ -256,6 +256,21 @@ def can_create_events(self, user):
         """Check whether the user can create events in the category."""
         return user and (not self.event_creation_restricted or self.can_manage(user, role='create'))
 
+    def move(self, target):
+        """Move the category into another category."""
+        assert not self.is_root
```

Comment:
You mean getting a cycle between categories? That's checked using a constraint trigger. I'd rather have a user-friendly check for this in the RH (or even better, disallow it on the widget level when selecting the move target)